orderer/common/multichannel/registrar.go

```diff
@@ -101,9 +105,27 @@ func NewRegistrar(
 		clusterDialer:      clusterDialer,
 	}
 
+	if config.ChannelParticipation.Enabled {
+		r.initializeJoinBlockFileRepo()
+	}
+
 	return r
 }
 
+// initialize the channel participation API joinblock file repo. This creates
+// the fileRepoDir on the filesystem if it does not already exist.
+func (r *Registrar) initializeJoinBlockFileRepo() {
+	fileRepoDir := filepath.Join(r.config.FileLedger.Location, "filerepo")
+	logger.Infof("Channel Participation API enabled, registrar initializing with file repo %s", fileRepoDir)
+
+	joinBlockFileRepo, err := filerepo.New(fileRepoDir, "joinblock")
+	if err != nil {
+		logger.Panicf("Error initializing join block file repo: %s", err)
```

Comment:
Shouldn't panic happen at the `main()` rather than here? Given that the existing code in this file already has some 15 panic statement, I am not going to block this PR on that. However, please consider returning the err to the caller and take the panic decision at the `main()`